src/sentry/utils/pubsub.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import absolute_import
+
+import redis
+import logging
+import random
+
+from django.conf import settings
+from threading import Thread
+from six.moves.queue import Queue
+
+
+class PubSub():
+    """
+    Poster for a redis pubsub instance.
+
+    put(channel, data) to send data to a redis pubsub
+    channel. Maintains an internal queue for posting, will discard the
+    value if the queue is full or not immediately available.
+    """
+
+    def __init__(self):
+        self._started = False
+
+    def _start(self):
+        if self._started:
+            return True
+
+        connection = getattr(settings, 'PUBSUB_CONNECTION', None)
+        try:
+            host, port, db = connection
+        except (TypeError, ValueError):
+            return False
+        self.rds = redis.StrictRedis(
+            host=host,
+            port=port,
+            db=db,
+            socket_timeout=0.2,
+            socket_connect_timeout=1,
+        )
+
+        self.q = q = Queue(maxsize=100)
+
+        def worker():
+            while True:
+                (channel, data) = q.get()
+                try:
+                    self.rds.publish(channel, data)
```

Comment:
Will this end up reconnecting if the connection was severed after it was established? I'm guessing it does, but just want to make sure.

Reply:
I can't actually find the redis-py documentation that says this is so, but I did just empirically test locally that it will automatically reconnect on the next `publish()` if the connection is killed / the server is reset.

Reply:
:+1: